### ISSUE TITLE
intermittent freebsd test_threads failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7574]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#341|
|Type|defect|
|Created|2003-10-16 00:38:20Z|

<details><summary>Searchable metadata</summary>

```
trac-id__341 341
type__defect defect
reporter__warner warner
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1066264700000000 1066264700000000
changetime__1468713621862448 1468713621862448
version__ 
owner__ 
cc__spiv cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
I've seen this happen once, on the freebsd builder.

FAILURE: testCallInThread (twisted.test.test_threads.ThreadsTestCase)
-------------------------------------------------------------------------------
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/test/test_threads.py",
line 90, in testCallInThread
    self.fail("threads lost a count")
FailTest: threads lost a count
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
This has started to happen more:
http://twistedmatrix.com/buildbot/freebsd/events/793/summary
http://twistedmatrix.com/buildbot/freebsd/events/788/summary
http://twistedmatrix.com/buildbot/freebsd/events/758/summary
http://twistedmatrix.com/buildbot/freebsd/events/753/summary

That's four out of the ten I currently see in the buildbot, which means at
a 40% failure rate, it has started to be the #1 reason for freebsd
test failures. It would be nice if some freebsd guy managed to fix that.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
There's only one occurance on the current buildbot page:
    http://twistedmatrix.com/buildbot/freebsd/events/982/summary

(that's the *only* red box in the entire visible history, for all slaves!  Woohoo!)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
So it's moved up to 100% :)
Seriously, can some fbsd guy look into this?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Uh, that's one occurrence out of nine visible FreeBSD buildslave runs... that's
hardly 100% :)

But it certainly looks like a reproducible problem.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
100% of reasons for fbsd failures? yes it is.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Ah, I misunderstood you :)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I recently added more information to the failure.
Looks more like thread starvation rather than synchronization:
it has 642/1000, which would mean that it has race condition half
the time. OTOH, if we assume freebsd starved threads for a little
over five seconds, we would have a problem.
Another option is that it's not starvation but deadlock issues,
but this seems unlikely...

FAILURE: testCallInThread (twisted.test.test_threads.ThreadsTestCase)
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
252, in _runPhase
    stage(*args, **kwargs)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
274, in _main
    self.runner(self.method)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/runner.py", line
102, in runTest
    method()
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/test/test_threads.py",
line 90, in testCallInThread
    self.fail("threads lost a count, index is %d "
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
98, in fail
    raise FailTest, message
FailTest: threads lost a count, index is 642  time is 1067501760.74, when is
1067501755.66
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Are these still happening?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
More than 6 months since I last asked, and nobody has re-reported.  I'm
resolving until someone decides to care about FreeBSD again.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_12"></a>|GitHub commented|
|-|-|

In changeset fa8f3cda35c0f68ec9df266b409d8fd8af45ddf4
```
#!CommitTicketReference repository="" revision="fa8f3cda35c0f68ec9df266b409d8fd8af45ddf4"
Merge pull request #341 from twisted/8609-rodrigc-win

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8609

Add _dumbwin32proc, _pollingfile, _win32stdio to fix importing on Python 3
```
